ang/civicase-base.ang.php

```diff
@@ -14,11 +14,21 @@
 use CRM_Civicase_Helper_NewCaseWebform as NewCaseWebform;
 use CRM_Civicase_Service_CaseCategoryCustomFieldsSetting as CaseCategoryCustomFieldsSetting;
 use CRM_Civicase_Helper_CaseUrl as CaseUrlHelper;
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
 
 [$caseCategoryId, $caseCategoryName] = CaseUrlHelper::getCategoryParamsFromUrl();
 
 $caseCategorySetting = new CRM_Civicase_Service_CaseCategorySetting();
 
+if (CRM_Utils_System::currentPath() !== 'civicrm/case/contact-case-tab') {
+  $notTranslationPath = $caseCategoryName == CaseCategoryHelper::CASE_TYPE_CATEGORY_NAME && CRM_Utils_System::currentPath() != 'civicrm/case/a';
+  if (!$notTranslationPath) {
+    if (!in_array($caseCategoryName, CaseCategoryHelper::getAccessibleCaseTypeCategories())) {
+      throw new Exception('Access denied! You are not authorized to access this page.');
+    }
+  }
+}
+
```

Comment:
Can we move the logic here https://github.com/compucorp/uk.co.compucorp.civicase/blob/ef7776cc6f9c6d8490d2fbf0c1e96c737eb83b5b/ang/civicase.ang.php#L22-L21 to this file also?
instead of having the logic in two places. We can have something like 

```php
if (CRM_Utils_System::currentPath() !== 'civicrm/case/contact-case-tab') {
  $notTranslationPath = $caseCategoryName == CaseCategoryHelper::CASE_TYPE_CATEGORY_NAME && CRM_Utils_System::currentPath() != 'civicrm/case/a';
  if (!$notTranslationPath) {
    if (!in_array($caseCategoryName, CaseCategoryHelper::getAccessibleCaseTypeCategories())) {
      throw new Exception('Access denied! You are not authorized to access this page.');
    }
    CRM_Civicase_Hook_Helper_CaseTypeCategory::addWordReplacements($caseCategoryName);
  }
}
```

or what do you think?